### PR TITLE
Development

### PR DESCRIPTION
**Pull request by VishnuPrasad-1995**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -6,3 +6,4 @@ spring.cloud.discovery.enabled = true
 server.port = 3005
 spring.data.mongodb.database=user_db
 spring.data.mongodb.port=27017
+logging.file.name=D:/Spring/Assessment/user-service/src/logs/user_service.log
```
Comment (Ramithags) [inline src/main/resources/application.properties]: So your code will logs to your local only please avoid this and try to give relative path

File: src/main/java/com/mavericsystems/userservice/service/UserServiceImpl.java [hunk 2]
```diff
@@ -19,21 +24,106 @@
     UserRepo userRepo;
 
     @Override
-    public UserDto createUser(User user) {
-        User user1 =  userRepo.save(user);
+    public UserDto createUser(UserRequest userRequest) {
+        if(userRepo.findByEmail(userRequest.getEmail()).isPresent()){
+            throw new EmailAlreadyExistException(EMAILALREADYEXIST);
+        }
+        User user1 = new User();
```
Comment (Ramithags) [inline src/main/java/com/mavericsystems/userservice/service/UserServiceImpl.java]: why are we calling this as user1 for objectName ?

File: src/main/java/com/mavericsystems/userservice/service/UserServiceImpl.java [hunk 3]
```diff
@@ -19,21 +24,106 @@
...
+        Optional<User> user = userRepo.findByEmail(emailId);
+
+        if (user.isPresent()) {
+            User user1 = user.get();
+            return new UserDto(user1.getId(), user1.getFirstName(), user1.getLastName(), user1.getMiddleName(), user1.getPhoneNumber(), user1.getEmail(), user1.getDateOfBirth(), user1.getEmployeeNumber(), user1.getBloodGroup(), user1.getGender());
+        }
+        else{
+            throw new UserNotFoundException(USERNOTFOUND + emailId);
+        }
+    }
```
Comment (Ramithags) [inline src/main/java/com/mavericsystems/userservice/service/UserServiceImpl.java]: remove unwanted space

File: src/main/java/com/mavericsystems/userservice/service/UserServiceImpl.java [hunk 4]
```diff
@@ -19,21 +24,106 @@
...
 
+    @Override
+    public UserDto getUserDetailsByEmail(String emailId) {
+        Optional<User> user = userRepo.findByEmail(emailId);
+
+        if (user.isPresent()) {
+            User user1 = user.get();
+            return new UserDto(user1.getId(), user1.getFirstName(), user1.getLastName(), user1.getMiddleName(), user1.getPhoneNumber(), user1.getEmail(), user1.getDateOfBirth(), user1.getEmployeeNumber(), user1.getBloodGroup(), user1.getGender());
+        }
+        else{
```
Comment (Ramithags) [inline src/main/java/com/mavericsystems/userservice/service/UserServiceImpl.java]: Please format the code base

File: src/main/java/com/mavericsystems/userservice/service/UserService.java [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
 import java.util.List;
 
 
 public interface UserService {
 
-    UserDto createUser(User user);
-
-    List<UserDto> getUsers();
-
+    UserDto createUser(UserRequest userRequest);
```
Comment (Ramithags) [inline src/main/java/com/mavericsystems/userservice/service/UserService.java]: Do you want to specify as the UserDto as a naming convention ? can you rename this by avoiding Dto

File: src/main/java/com/mavericsystems/userservice/dto/UserDto.java [hunk 6]
```diff
@@ -17,9 +18,8 @@
     private String middleName;
     private String phoneNumber;
```
Comment (Ramithags) [inline src/main/java/com/mavericsystems/userservice/dto/UserDto.java:19]: Please add regex validation

File: src/main/java/com/mavericsystems/userservice/dto/UserRequest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    @NotNull(message = "Middle name should not null")
+    private String middleName;
+    @NotBlank(message = "Phone number must not be blank")
+    @NotNull(message = "Phone number should not null")
+    @Size(min = 10, max = 10, message = "phone number should be a 10 digit number")
+    private String phoneNumber;
+    @NotBlank(message = "email should not be blank")
+    @NotNull(message = "email should not be null")
+    @Email(message = "Please enter a valid email address")
+    private String email;
```
Comment (Ramithags) [inline src/main/java/com/mavericsystems/userservice/dto/UserRequest.java:33]: Add regex validation pattern

File: src/main/java/com/mavericsystems/userservice/controller/UserController.java [hunk 8]
```diff
@@ -19,19 +21,37 @@
...
     @PostMapping
-    public ResponseEntity<UserDto> createUser(@Valid @RequestBody User user){
+    public ResponseEntity<UserDto> createUser(@Valid @RequestBody UserRequest userRequest){
+
         try {
-            return new ResponseEntity<>(userService.createUser(user), HttpStatus.CREATED);
+            return new ResponseEntity<>(userService.createUser(userRequest), HttpStatus.CREATED);
         }
         catch (Exception e){
             throw new CustomCreateUserException("Syntax Error");
```
Comment (Ramithags) [inline src/main/java/com/mavericsystems/userservice/controller/UserController.java]: Is it a syntax error ? are you sure

File: src/main/java/com/mavericsystems/userservice/constant/UserConstant.java [hunk 9]
```diff
@@ -6,5 +6,8 @@ private UserConstant() {
         // restrict instantiation
     }
 
-    public final static String deleteCustomer = "User successfully deleted"  ;
+    public static final String DELETEUSER = "User successfully deleted"  ;
+    public static final String USERNOTFOUND = "User not found with id : ";
+    public static final String NOUSERFOUND = "No user data available : ";
+    public static final String EMAILALREADYEXIST = "Email Already exist : ";
```
Comment (Ramithags) [inline src/main/java/com/mavericsystems/userservice/constant/UserConstant.java]: Follow naming convention guide if possible https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

**Review by Ramithags — CHANGES_REQUESTED** [review]

**Review by Ramithags — COMMENTED** [review]
Can you please write unit tests

**Comment by VishnuPrasad-1995** [comment]
Changes are made as per reviewer, New pull request will be raised with modifications. So this PR is closed

**Pull request closed without merging**